src/OpenTelemetry.Extensions.Hosting/Implementation/TracerProviderBuilderHosting.cs

```diff
@@ -109,6 +109,24 @@ public TracerProvider Build(IServiceProvider serviceProvider)
             return this.Build();
         }
 
+        IDeferredTracerProviderBuilder IDeferredTracerProviderBuilder.AddService(Type serviceType, object instance)
+        {
+            this.Services.Add(new ServiceDescriptor(serviceType, instance));
+            return this;
+        }
+
+        IDeferredTracerProviderBuilder IDeferredTracerProviderBuilder.AddService(Type serviceType, Type implementationType, ServiceLifetime lifetime)
+        {
+            this.Services.Add(new ServiceDescriptor(serviceType, implementationType, (Microsoft.Extensions.DependencyInjection.ServiceLifetime)(int)lifetime));
```

Comment:
Although it's highly unlikely for the `ServiceLifetime` enum's integral types to change, do we want to be extra safe by adding an `Enum.IsDefined` check before casting the enum?

`Enum.IsDefined(typeof(Microsoft.Extensions.DependencyInjection.ServiceLifetime.ServiceLifetime), (int)lifetime)`.

Alternatively, we could also just add a method with switch-case construct to use the actual enumeration type instead of the integral type to return the correctly mapped enum. Something like this:

```
internal GetServiceLifeTime(ServiceLifeTime serviceLifetime)
    => serviceLifetime switch
        {
            ServiceLifeTime.Singleton => Microsoft.Extensions.DependencyInjection.ServiceLifetime.Singleton,
            ServiceLifeTime.Scoped => Microsoft.Extensions.DependencyInjection.ServiceLifetime.Scoped,
            ServiceLifeTime.Transient => Microsoft.Extensions.DependencyInjection.ServiceLifetime.Transient,
            _ => throw new InvalidEnumArgumentException(nameof(serviceLifetime), (int)serviceLifetime, typeof(Microsoft.Extensions.DependencyInjection.ServiceLifetime))
        };
```

This might be an overkill 😄

Reply:
Totally overkill but I incorporated it :smile: